### ISSUE TITLE
Future ggplot2 3.5.0 breaks geom_label_s() and geom_label()pairwise

### ISSUE THREAD

**Issue by aphalo** [aphalo/ggpp#45]
In addition to the expected small change in boxes in those geoms that call ggplot2::GeomLabel  these to geoms that generate their own grobTrees silently display only the segments. I suspect an internal change in 'ggplot2' affecting the construction or addition of the grobs to the plot. However, I need to check what is going on.

**Comment by aphalo** [comment]
Copied from https://github.com/tidyverse/ggplot2/issues/5419
Hi Pedro,

    Has anything changed in 'ggplot2' in how grobs (a grobTree in this case) are added?

For geom_label() we've changed the implementation of how the drawing functions work because we wanted to support an angle aesthetic (among others). More details in https://github.com/tidyverse/ggplot2/pull/5030. For {ggpp} a quick glance gives me the intuition that in the lines below, the cl = "labelgrob" is not doing what it used to.

https://github.com/aphalo/ggpp/blob/175b38f54a73f66b7e44ef58121f51f23e55f6d6/R/geom-label-linked.r#L262-L263C94

We simplified the class away because we saw a simpler solution that didn't need the grid::makeContent methods for this class.

    Second question, is there already some schedule for the release of 'ggplot2' 3.5.0?

Unfortunately I don't have a detailed grasp on this, but a meatier, non-hotfix release might come out sometime this year.

    Ideally, there would be solution to this problem, that would work with 3.4.0 and earlier as well as 3.5.0.

I suspect that the development labelGrob() implementation would also fit with previous versions, so if you're copying the grob constructor, I'd try the new one.
